src/main/java/com/google/api/codegen/config/DiscoveryField.java

```diff
@@ -23,36 +23,89 @@
 import com.google.api.codegen.discovery.Schema.Type;
 import com.google.api.codegen.transformer.ImportTypeTable;
 import com.google.api.codegen.util.Name;
+import com.google.api.codegen.util.SymbolTable;
 import com.google.api.codegen.util.TypeName;
 import com.google.api.tools.framework.model.Oneof;
 import com.google.api.tools.framework.model.TypeRef.Cardinality;
 import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
 import com.google.common.collect.ImmutableList;
+import java.util.AbstractMap;
+import java.util.Comparator;
 import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 import javax.annotation.Nullable;
 
 /** A field declaration wrapper around a Discovery Schema. */
 public class DiscoveryField implements FieldModel, TypeModel {
   private final List<DiscoveryField> properties;
+  private final DiscoApiModel apiModel;
   // Dereferenced schema to use for rendering type names and determining properties, type, and format.
   private final Schema schema;
-
   // Not dereferenced schema; used in rendering this FieldModel's parameter name.
   private final Schema originalSchema;
-  private final DiscoApiModel apiModel;
+
+  // Unformatted type name for this Field. For message-type Fields, this will be globally unique.
```

Comment:
Shouldn't this be unique per-package (namespace)?

Reply:
Great point! Yes indeed.